code/modules/mob/living/carbon/give.dm

```diff
@@ -1,52 +1,184 @@
-/mob/living/carbon/verb/give(mob/living/carbon/target in oview(1))
-	set category = "IC"
-	set name = "Give"
+/**
+ * Toggles the [/datum/click_intercept/give] on or off for the src mob.
+ */
+/mob/living/carbon/proc/toggle_give()
+	if(has_status_effect(STATUS_EFFECT_OFFERING_ITEM))
+		to_chat(src, "<span class='warning'>You're already offering an item to someone!</span>")
```

Comment:
Design thing:
Maybe this should cancel the active offering instead?

Reply:
I'm not sure how I feel about this. However, what I will do make it so if the giver clicks the offering item alert, it cancels the offer.